xaitk_saliency/impls/perturb_image/sliding_radial.py

```diff
@@ -0,0 +1,84 @@
+from typing import Any, Dict, Tuple
+import numpy as np
+from xaitk_saliency.interfaces.perturb_image import PerturbImage
+from scipy.ndimage.filters import gaussian_filter
+
+
+class SlidingRadial (PerturbImage):
+    """
+    Produce perturbation matrices generated by sliding a circle occlusion area
+    with a configured radius over the area of an image. Passing a sigma value
+    will apply a Gaussian filter to the mask, blurring it. This results in a
+    smooth transition from full occlusion in the center of the radial to no
+    occlusion at the edge.
+
+    Due to the geometry of sliding windows, if the stride given does not evenly
+    divide the window size along the applicable axis, then the result plane of
+    values when summing the generated masks will not be even.
+
+    Related, if the stride is set to be larger than the circle diameter, the
+    resulting plane of summed values will also not be even, as there be
+    increasingly long valleys of unperturbed space between masked regions.
+
+    :param radius: The radius of the occlusion area.
+    :param sigma: The sigma value for the Gaussian filter applied to mask.
+    :param stride: The striding step for the center of the radial as a tuple
+        with format `(height_step, width_step)`.
+    """
+
+    def __init__(
+        self,
+        radius: int = 50,
+        sigma: int = 0,
+        stride: Tuple[int, int] = (20, 20),
+    ):
+        self.radius: int = int(radius)
+        self.sigma: int = int(sigma)
+        self.stride: Tuple[int, int] = (int(stride[0]), int(stride[1]))
+
+    def perturb(
+        self,
+        ref_image: np.ndarray
+    ) -> np.ndarray:
+        stride_h, stride_w = self.stride
+        img_size = ref_image.shape[:2]
+        img_h, img_w = img_size
+        center_xs = np.arange(0, img_w, stride_w)
```

Comment:
Given a previous conversation, do we want this to start with the center of the circle in the upper left? Or with the circle fully within the image (e.g. `np.arange` start = `radius`).

Reply:
I think we should revisit this and make a similar design choice for SlidingWindow (i.e. whether the center of the box/circle starts at 0,0 or fully includes the box/circle).